gcp-observability/src/test/java/io/grpc/gcp/observability/MetricsTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.gcp.observability;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.monitoring.v3.MetricServiceClient;
+import com.google.cloud.monitoring.v3.MetricServiceClient.ListTimeSeriesPagedResponse;
+import com.google.monitoring.v3.ListTimeSeriesRequest;
+import com.google.monitoring.v3.ProjectName;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TimeSeries;
+import com.google.protobuf.util.Timestamps;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.gcp.observability.interceptors.InternalLoggingChannelInterceptor;
+import io.grpc.gcp.observability.interceptors.InternalLoggingServerInterceptor;
+import io.grpc.gcp.observability.logging.GcpLogSink;
+import io.grpc.gcp.observability.logging.Sink;
+import io.grpc.testing.GrpcCleanupRule;
+import io.grpc.testing.protobuf.SimpleServiceGrpc;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class MetricsTest {
+
+  private static final String PROJECT_ID = "PROJECT";
+  private String customTagKey = "Version";
+  private String customTagValue =
+      String.format("C67J9A-%s", String.valueOf(System.currentTimeMillis()));
+  private Map<String, String> customTags = Collections.singletonMap(customTagKey, customTagValue);
+
+  @Rule public final GrpcCleanupRule cleanupRule = new GrpcCleanupRule();
+
+  /**
+   * Cloud Monitoring test using GlobalInterceptors.
+   *
+   * <p>Ignoring test, because it calls external Cloud Monitoring APIs. To test cloud monitoring
+   * setup locally, 1. Set up Cloud auth credentials 2. Assign permissions to service account to
+   * write metrics to project specified by variable PROJECT_ID 3. Comment @Ignore annotation
```

Comment:
I am assuming when enabled with the correct set up this test passes? It might be worth mentioning that.